src/uct/tcp/tcp_iface.c

```diff
@@ -943,6 +943,15 @@ int uct_tcp_keepalive_is_enabled(uct_tcp_iface_t *iface)
 #endif /* UCT_TCP_EP_KEEPALIVE */
 }
 
-UCT_TL_DEFINE(&uct_tcp_component, tcp, uct_tcp_query_devices, uct_tcp_iface_t,
-              UCT_TCP_CONFIG_PREFIX, uct_tcp_iface_config_table,
-              uct_tcp_iface_config_t);
+UCT_COMPONENT_REGISTER_DEF(&uct_tcp_component, tcp)
+UCT_TL_REGISTER_DEF(&uct_tcp_component, tcp, uct_tcp_query_devices, uct_tcp_iface_t,
+                    UCT_TCP_CONFIG_PREFIX, uct_tcp_iface_config_table,
+                    uct_tcp_iface_config_t);
+
+UCT_TL_INIT(tcp)
```

Comment:
do we really need to define empty ctor/dtors?

Reply:
xpmem needs custom steps to initialize, used common macro for this

Reply:
what if we just remove this line for TCP and SELF?

Reply:
> what if we just remove this line for TCP and SELF?

not sure I got your point...
we can't call component/tl ctors/dtors directly because it declared as static routines and ```UCT_TL_INIT``` macro is used to provide single entry point to initialise tl/component, empty custom body is excluded by compiler because it is declared as static routine too.

but ok, replaced by regular routine

Reply:
oh.. i missed that UCT_TL_INIT is actually doing some stuff, and the "empty function body" here just means it does not do any "extra" stuff
so better revert to using UCT_TL_INIT

Reply:
> oh.. i missed that UCT_TL_INIT is actually doing some stuff, and the "empty function body" here just means it does not do any "extra" stuff
> so better revert to using UCT_TL_INIT

reverted